[FEAT] banner 생성/수정/삭제 api 추가 및 banner service 로직 분리

mkshie opened the pull request:
## 개요
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
<!---- Resolves: #(Issue Number) -->

- 메인 배너 관리를 위한 **도메인·매퍼·서비스 구조**를 정리하고,
- **관리자용/사용자용 배너 조회 API**를 분리,
- **배너 노출 순서 정렬 API**를 추가했습니다.
- 추후 서브배너 확장까지 고려해 enum/필드 설계를 정리했습니다.

## 주요 변경 사항

### 1. 배너 도메인 및 매퍼 정리
- `EventBanner` 전용 `BannerMapper` 분리
  - `toCreateBannerResponse`, `toEventBanner` 등 엔티티 ↔ DTO 변환 책임 분리
- 필드/기본값 정리
  - `selected` 기본값 추가 (기본 노출 여부)
  - `startAt`, `endAt` 타입을 `LocalDate` 로 변경해 **일 단위 노출 기간 관리**가 명확해지도록 수정
- 컨트롤러/서비스/레포/enum 정리
  - 배너 생성/수정/삭제 API 컨트롤러 정리
  - `BannerType`, `BANNER_ENTITY_NOT_FOUND` 등 배너 전용 enum 추가

### 2. 배너 조회 API 세분화
- **관리자용 조회 API (`getEventBanners`)**
  - 아래 세 가지 그룹을 한 번에 조회하는 응답 구조 설계
    - 현재 화면에 노출 중인 배너
    - 노출 시작 전(대기) 배너
    - 이미 노출이 끝난 이전 배너
  - 이를 표현하기 위한 `EventBannerAdminResponse` 도입
  - 기간 기준 조회를 위해 레포에 메서드 추가  
    - `findPastEventBannersByType(...)`  
    - `findCurrentAndWaitingEventBannersByType(...)`
- **메인 화면용 조회 API (`getActiveEventBanners`)**
  - 현재 시점 기준, 노출 조건을 만족하는 배너만 반환
  - `displayOrder` 기준으로 정렬해 **UI에서 바로 사용 가능한 순서**로 전달
- 응답 포맷
  - `@JsonInclude(Include.NON_NULL)` 적용

### 3. 배너 순서 정렬 API 추가
- `BannerOrderUpdateRequest` 추가
  - 클라이언트에서 `배너 ID 리스트`를 넘기면, 해당 순서대로 `displayOrder` 재정렬
- 검증 로직
  - `EventBannerRepository.findByIdIn(...)` 으로 **전달된 ID 존재 여부 확인**
  - 노출 기간이 유효하지 않은 배너에 대해서는 정렬 대상에서 제외/에러 처리
  - 잘못된 ID에 대해서는 `EventErrorCode.INVALID_BANNER_ID` 응답
- 추후 서브배너가 추가될 것을 고려해, `bannerType` 은 유지한 채 로직만 보완

### 4. 기타 리팩토링 및 더미 데이터
- 서브배너 타입 제거로 인해 깨진 기존 로직 보완 
- 배너 관련 서비스/매퍼 코드 정리 및 불필요한 import 제거
- 로컬/테스트용으로 배너 더미 데이터 추가

## PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.  Commit message convention 참고  (Ctrl + 클릭하세요.)
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트)

mkshie committed:
feat(banner) : BannerMapper 분리

- toCreateBannerResponse ,toEventBanner(entity 반환) 생성

mkshie committed:
feat(banner) : selected 디폴트값 추가 , startAt 및 endAt LocalDate 형식으로 변경, updateInfo 메소드 추가 , updateBannerImage 메소드 추가

mkshie committed:
feat(banner) : 컨트롤러 , 서비스 , 레포 수정 ,enum 추가 , request 추가

- 컨트롤러 : 생성 , 삭제 , 수정 api 추가
- 서비스 :  위와 동일
- 레포 : findTopByTypeOrderByDisplayOrderDesc 추가
- enum : BANNER_ENTITY_NOT_FOUND 추가

mkshie committed:
fix(banner) : BannerMapper 분리

- toCreateBannerResponse ,toEventBanner(entity 반환) 생성

mkshie committed:
fix(banner) : Service 분리 , 불필요한 import 문 제거 , 코드 재정렬 (수정사항 x)

mkshie committed:
fix(bannerResponse) : @JsonInclude(JsonInclude.Include.NON_NULL) 추가  LocalDate 로 형식 수정

kimjunghwannn reviewed (commented):
깔끔하게 잘 짜셨네요 고생하셨습니다!

mkshie committed:
fix(banner) : subbanner 가 사라져서 기존 로직 수정

- main 배너는 추후 서브배너가 생길 수 있으니 bannerType 은 남겨두고 작업했습니다

mkshie committed:
feat(banner) : banner 순서 정렬 api 작성
- EventErrorCode 에 INVALID_BANNER_ID 추가
- BannerOrderUpdateRequest 추가
- EventBannerRepository 에 findByIdIn 추가
- EventBannerService 에서 존재하는 배너인지 확인 , 기간이 올바른 배너인지 확인 후 순서 재정렬

mkshie committed:
fix(banner) : 배너 조회 API 디자인에 맞게 세분화
- 관리자용 배너 조회 API(getBannersAll) 와 메인화면 배너 조회 API(getHomeBanners) 분리
- 관리자용 배너조회(getEventBanners) 의 경우 화면에 뜨는 배너 + 아직 뜨지않은 배너 / 이전에 사용했던 배너를 조회
- 메인화면용 배너조회(getActiveEventBanners) 의 경우 화면에 뜨는 배너만 조회
- EventBannerAdminResponse 추가
- repo 에  findPastEventBannersByType(이전배너) 과 findCurrentAndWaitingEventBannersByType(아직 + 화면에 뜨는 배너) 추가

mkshie committed:
feat(data) : 배너용 데이터 추가

mkshie commented:
다시 수정했습니다!

kimjunghwannn reviewed (commented):
깔끔하게 잘 하셨네요!
하나만 말씀드릴게 있는데 다른 도메인 페이지 번호가 다 0번 부터여서 페이지 번호를 0번부터로 수정 하는게 좋아보입니다!

mkshie commented:
그 부분 수정한거 지금 작업중인거랑 같이 pr 날릴게요!
